docs/performance/transformers-optimization.md

```diff
@@ -0,0 +1,266 @@
+* Experimenting with disabling or enabling some fusions to evaluate impact on performance or accuracy.
+
+### Usage workflow
+1. [Install ONNX Runtime](#1-install-onnx-runtime)
+2. [Convert the transformer model to ONNX](#2-convert-a-transformer-model-to-onnx)
+3. [Run the model optimizer tool](#3-run-the-model-optimizer-tool)
+4. [Benchmark and profile the model](#4-benchmark-and-profile-the-model)
+
+### Supported models
+
+Below is the list of models that have been explicitly tested with the optimizer. Models not in the list may only be partially optimized or not optimized at all.
+
+PyTorch (from [Huggingface Transformers](https://github.com/huggingface/transformers/)):
+- BERT
+- DistilBERT
+- DistilGPT2
+- RoBERTa
+- ALBERT
+- GPT-2 (**GPT2Model**, **GPT2LMHeadModel**)
+
+TensorFlow
+- The only TensorFlow model tested so far is BERT.
+
+Most optimizations require exact match of a subgraph. Any layout change in the subgraph might cause some optimization to not work. Note that different versions of training or export tool might lead to different graph layouts. It is recommended to use the latest released version of PyTorch and Transformers.
+
+#### Limitations to note
+* Due to the CUDA implementation of the Attention kernel in ONNX Runtime, the maximum number of attention heads is 1024. 
+* Normally, due to GPU memory constraints, the maximum supported sequence length is 4096 for Longformer and 1024 for other types of models.
+
+---
+
+## 1. Install ONNX Runtime
+
+First you need install onnxruntime or onnxruntime-gpu package for CPU or GPU inference. To use onnxruntime-gpu, it is required to install CUDA and cuDNN and add their bin directories to PATH environment variable. See [Python installation instructions](./../install/index.md#python-installs).
+
+
+## 2. Convert a transformer model to ONNX
+
+To convert the transformer model to ONNX, use [torch.onnx](https://pytorch.org/docs/stable/onnx.html) or [tensorflow-onnx](https://github.com/onnx/tensorflow-onnx). 
+
+- Huggingface transformers has a [notebook](https://github.com/huggingface/notebooks/blob/master/examples/onnx-export.ipynb) shows an example of exporting a pretrained model to ONNX.
+
+- For tf2onnx, please refer to this [BERT tutorial](https://github.com/onnx/tensorflow-onnx/blob/master/tutorials/BertTutorial.ipynb). 
+
+### GPT-2 Model conversion
+
+Converting the GPT-2 model from PyTorch to ONNX is not straightforward when past state is used. The tool [convert_to_onnx](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/convert_to_onnx.py) can help.
+
+You can use commands like the following to convert a pre-trained PyTorch GPT-2 model to ONNX for given precision (float32, float16 or int8):
+```
+python -m onnxruntime.transformers.convert_to_onnx -m gpt2 --model_class GPT2LMHeadModel --output gpt2.onnx -p fp32
```

Comment:
This section need update. We can remove int8 example since no one is using it.

```
python -m onnxruntime.transformers.models.gpt2.convert_to_onnx -m gpt2 --model_class GPT2LMHeadModel --output gpt2.onnx -p fp32

python -m onnxruntime.transformers.models.gpt2.convert_to_onnx -m distilgpt2 --model_class GPT2LMHeadModel --output distilgpt2.onnx -p fp16 --use_gpu --optimize_onnx --auto_mixed_precision
```